app/javascript/controllers/receipt_input_controller.js

```diff
@@ -0,0 +1,75 @@
+import { Controller } from '@hotwired/stimulus'
+import csrf from '../common/csrf'
+
+export default class extends Controller {
+  static targets = ['list', 'fileInput', 'clearButton', 'bin', 'receiptsInput']
+  static outlets = ['receipt-select', 'extraction']
+
+  binOpen = false
+
+  receiptSelectOutletConnected(outlet) {
+    outlet.render()
+  }
+
+  addFile() {
+    const files = this.fileInputTarget.files
+    const newFile = files[files.length - 1]
+
+    const html = `<li>${newFile.name}</li>`
+    this.listTarget.innerHTML += html
+    this.clearButtonTarget.classList.remove('hidden')
```

Comment:
I think may want to work on the styling a bit more. The list feels a little out of place at the moment.

Reply:
worked on a new list that incorporates the file preview (using the payment-docs icon as fallback when there's no preview) - what do you think?
<img width="547" height="263" alt="image" src="https://github.com/user-attachments/assets/04f8bf3f-ff22-49d3-86b3-42a7bf56eb1e" />